synthesizer/process/src/stack/finalize_types/initialize.rs

```diff
@@ -16,6 +16,29 @@
 use super::*;
 
 impl<N: Network> FinalizeTypes<N> {
+    /// Initializes a new instance of `FinalizeTypes` for the given constructor.
+    /// Checks that the given constructor is well-formed for the given stack.
+    #[inline]
+    pub(super) fn initialize_finalize_types_from_constructor(
+        stack: &(impl StackMatches<N> + StackProgram<N>),
+        constructor: &Constructor<N>,
+    ) -> Result<Self> {
+        // Initialize a map of registers to their types.
+        let mut finalize_types = Self { inputs: IndexMap::new(), destinations: IndexMap::new() };
+
+        // Check the commands are well-formed.
+        for command in constructor.commands() {
+            // Verify that the command is not an await.
+            if command.is_await() {
```

Comment:
Are there any other finalize-specific commands or operands we should prevent from using? Are these fine? https://developer.aleo.org/guides/aleo/special_operands

Reply:
This is the only one.